tests/Aspire.Hosting.Tests/Dashboard/ResourcePublisherTests.cs

```diff
@@ -131,6 +131,35 @@ public async Task DeletesRemoveFromSnapshot()
         await cts.CancelAsync();
     }
 
+    [Fact]
+    public async Task CancelledSubscriptionIsCleanedUp()
+    {
+        ResourcePublisher publisher = new(CancellationToken.None);
+        CancellationTokenSource cts = new();
+        var called = false;
+
+        var (_, subscription) = publisher.Subscribe();
```

Comment:
Can you make _outgoingChannels internal and assert values are added and removed? That's the ultimate test that there isn't a leak.

Reply:
I'll extend these tests.